scheduler/cron.go

```diff
@@ -0,0 +1,198 @@
+package scheduler
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"strings"
+)
+
+type cron struct{}
+
+// Enable writes Fluidkeys' cron lines into crontab
+func (c *cron) Enable() (crontabWasAdded bool, err error) {
+	crontab := &systemCrontab{}
+	crontabWasAdded, err = c.enable(crontab)
+	if err != nil {
+		return crontabWasAdded, ErrModifyingCrontab{origError: err}
```

Comment:
The intent of the custom `ErrModifyingCrontab` error is to tell the caller there was a problem **modifying** the crontab, so they can go and do it manually.

By wrapping *all errors* from `enable` this specific error can now be raised for numerous reasons, some of which might not be resolved by manually editing your crontab.

Suggest putting `Enable` back to the simple one-liner and returning `ErrModifyingCrontab` from inside `enable`, *only* when it's appropriate.

Reply:
Yep - you're right.